### PR TITLE
ref(ts) Update OrganizationsStore to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

**Review by scttcper — APPROVED** [review]
looks good, might need to add loaded to the OrganizationsStoreInterface

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  getInitialState() {
+    return this.state;
+  },
+
+  init() {
+    this.state = [];
+    this.loaded = false;
+  },
+
+  onUpdate(org) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:37]:
```suggestion
  onUpdate(org: Organization) {
```

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    this.remove(prev.slug);
+    this.add(next);
+  },
+
+  onRemoveSuccess(slug: string) {
+    this.remove(slug);
+  },
+
+  get(slug) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:54]:
```suggestion
  get(slug: string) {
```

or 

```suggestion
  get(slug: Organization['slug']) {
```

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    return this.state.find((item: Organization) => item.slug === slug);
+  },
+
+  getAll() {
+    return this.state;
+  },
+
+  remove(slug) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:62]:
```suggestion
  remove(slug: string) {
```

or 

```suggestion
  remove(slug: Organization['slug']) {
```

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    this.state = this.state.filter((item: Organization) => slug !== item.slug);
+    this.trigger(this.state);
+  },
+
+  add(item) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:67]:
```suggestion
  add(item: Organization) {
```

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        match = true;
+      }
+    });
+    if (!match) {
+      this.state = [...this.state, item];
+    }
+    this.trigger(this.state);
+  },
+
+  load(items) {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:82]:
```suggestion
  load(items: Organization[]) {
```

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  get(slug) {
+    return this.state.find((item: Organization) => item.slug === slug);
+  },
+
+  getAll() {
+    return this.state;
+  },
+
+  remove(slug) {
+    this.state = this.state.filter((item: Organization) => slug !== item.slug);
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:63]:
```suggestion
    this.state = this.state.filter(item => slug !== item.slug);
```

image

File: src/sentry/static/sentry/app/stores/organizationsStore.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    this.trigger(this.state);
+  },
+
+  add(item) {
+    let match = false;
+    this.state.forEach((existing: Organization, idx: number) => {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/stores/organizationsStore.tsx:69]:
```suggestion
    this.state.forEach((existing, idx) => {
```

image

**Comment by priscilawebdev** [comment]
it looks good...I left just a few suggestions 😉

**Pull request merged**